core/src/main/scala/zio/jdbc/package.scala

```diff
@@ -15,9 +15,12 @@
  */
 package zio
 
+import zio.jdbc.SqlFragment.{ Segment, Setter }
+import zio.jdbc.{ JdbcEncoder, SqlFragment }
+
 import scala.language.implicitConversions
 
-package object jdbc {
+package object jdbc extends LowPriorityImplicits1 {
```

Comment:
It's not great practice to use package objects because of their removal in Scala 3. 

I think we can solve this problem in another way:

By using `JdbcEncoder`, and deleting `Setter`.

In order to do this, we would enhance `JdbcDecoder` with two new methods like so:

```scala
def unsafeEncode(columIndex: Int, ps: PreparedStatement, value: A): Int
def unsafeEncodeNull(columIndex: Int, ps: PreparedStatement): Int
```

and implement them for all existing `JdbcEncoder`.

These are essentially `Setter#setValue` / `setNull`, except the functions return the output index, because it may require more than 1 index to write the full value. 

What do you think?

Reply:
I wonder if JdbcEncode and Setter could be combined under one sealed trait.

From what I understand Setter is a bit safer since it uses PreparedStatement.

I will try some experiments today.

Reply:
The main thing is if we really want to use PreparedStatements or do we want to do the inlining ourselves ..

Reply:
All right, I've combined Setter into JdbcDecoder and I think it looks pretty good.

It's not finished - there is some refactoring to do. But you can already check if the idea is fine in your opinion.